server/src/main/java/org/opensearch/indices/tiering/TieringRequestValidator.java

```diff
@@ -0,0 +1,297 @@
+     */
+    static boolean validateHotIndex(final ClusterState state, final Index index) {
+        return IndexModule.TieringState.HOT.name().equals(INDEX_TIERING_STATE.get(state.metadata().getIndexSafe(index).getSettings()));
+    }
+
+    /**
+     * Validates the health of the specified index in the current cluster state.
+     *
+     * @param currentState the current cluster state
+     * @param index the index to be validated
+     * @return true if the index health is not in the "red" state, false otherwise
+     */
+    static boolean validateIndexHealth(final ClusterState currentState, final Index index) {
+        final IndexRoutingTable indexRoutingTable = currentState.routingTable().index(index);
+        final IndexMetadata indexMetadata = currentState.metadata().index(index);
+        final ClusterIndexHealth indexHealth = new ClusterIndexHealth(indexMetadata, indexRoutingTable);
+        return !ClusterHealthStatus.RED.equals(indexHealth.getStatus());
+    }
+
+    /**
+     * Validates that the specified index is in the open state in the current cluster state.
+     *
+     * @param currentState the current cluster state
+     * @param index the index to be validated
+     * @return true if the index is in the open state, false otherwise
+     */
+    static boolean validateOpenIndex(final ClusterState currentState, final Index index) {
+        return currentState.metadata().index(index).getState() == IndexMetadata.State.OPEN;
+    }
+
+    /**
+     * Validates that the disk threshold low watermark is not breached on all the eligible nodes in the cluster.
+     *
+     * @param currentState the current cluster state
+     * @param clusterInfo the current nodes usage info for the cluster
+     * @param diskThresholdSettings the disk threshold settings of the cluster
+     * @param indexNames the names of the indices being validated
+     * @throws IllegalStateException if the disk threshold low watermark is breached on all eligible nodes
+     */
+    static void validateDiskThresholdWaterMarkNotBreached(
+        final ClusterState currentState,
+        final ClusterInfo clusterInfo,
+        final DiskThresholdSettings diskThresholdSettings,
+        final String indexNames
+    ) {
+        final Map<String, DiskUsage> usages = clusterInfo.getNodeLeastAvailableDiskUsages();
+        if (usages == null) {
+            logger.trace("skipping monitor as no disk usage information is available");
+            return;
+        }
+        final Set<String> nodeIds = getEligibleNodes(currentState).stream().map(DiscoveryNode::getId).collect(Collectors.toSet());
+        for (String node : nodeIds) {
+            final DiskUsage nodeUsage = usages.get(node);
+            if (nodeUsage != null && nodeUsage.getFreeBytes() > diskThresholdSettings.getFreeBytesThresholdLow().getBytes()) {
+                return;
+            }
+        }
+        throw new IllegalStateException(
```

Comment:
Lets convert this to `IllegalArgumentException` as well since otherwise we will send 5xx status code to user which could be confusing because we are failing the request as part of check instead of some server side execution error